skfp/preprocessing/filters/rule_of_4.py

```diff
@@ -0,0 +1,91 @@
+from typing import Optional
+
+from rdkit.Chem import Mol
+from rdkit.Chem.Crippen import MolLogP
+from rdkit.Chem.Descriptors import MolWt
+from rdkit.Chem.rdMolDescriptors import CalcNumHBA, CalcNumRings
+
+from skfp.bases.base_filter import BaseFilter
+
+
+class RuleOf4(BaseFilter):
+    """
+    Rule designed to look for molecules used as PPI (protein-protein inhibitor).
+    Described in [1].
+
+    Acording to this rule PPI compound should meet following criteria:
+    - molecular weight >= 400 daltons
+    - HBA >= 4
+    - logp >=4
+    - number of rings >= 3
+
+    Parameters
+    ------------
+    allow_one_violation : bool, default=True
+        Whether to allow violating one of the rules for a molecule. This makes the
+        filter less restrictive, and is the part of the original definition of this
+        filter.
+
+    n_jobs : int, default=None
+        The number of jobs to run in parallel. :meth:`transform_x_y` and
+        :meth:`transform` are parallelized over the input molecules. ``None`` means 1
+        unless in a :obj:`joblib.parallel_backend` context. ``-1`` means using all
+        processors. See Scikit-learn documentation on ``n_jobs`` for more details.
+
+    batch_size : int, default=None
+        Number of inputs processed in each batch. ``None`` divides input data into
+        equal-sized parts, as many as ``n_jobs``.
+
+    verbose : int, default=0
+        Controls the verbosity when generating conformers.
+
+    References
+    -----------
+    .. [1] 'Morelli, X., Bourgeas, R., & Roche, P. (2011).
+    Chemical and structural lessons from recent successes in protein–protein interaction inhibition (2P2I).
+    Current Opinion in Chemical Biology, 15(4), 475–481. doi:10.1016/j.cbpa.2011.05.024'_
+
+    Examples
+    ----------
+    >>> from skfp.preprocessing import RuleOf4
+    >>> smiles = ["c1ccc2oc(-c3ccc(Nc4nc(N5CCCCC5)nc(N5CCOCC5)n4)cc3)nc2c1", \
+    "c1nc(N2CCOCC2)c2sc3nc(N4CCOCC4)c4c(c3c2n1)CCCC4" ]
```

Comment:
Does this pass doctest? `make docs` will run this and print failures if necessary